checker.c

```diff
@@ -1,21 +1,102 @@
+/* Include Headerfiles													*/
+/* ===================													*/
+
 #include <stdio.h>
-#include <assert.h>
+#include <assert.h>	
+#include <stdbool.h>
+#include "checker.h"
+
+
+
+/********************************************************************************
+ * A common function that checks if the paramters are within the boundary conditions.
+ * input: parameter value, Maximum and minimum range to be checked
+ * returns: Check if the parameter is within in the given maximum and minimum range
+ *********************************************************************************/
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+bool BMS_AttributeinRangeStatus(float parametervalue, int index)
+
+{	bool InRange=true;
+	if ((parametervalue< (AttributeMinimumthreshold[index]))||(parametervalue>= (AttributeMaximumthreshold[index])))
+	  {		
+		  return (InRange=false);
+	  }
+	return(InRange);
 }
 
-int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
+/********************************************************************************
+ * This function accumulates the breach status for all the parameters of the BMS and stores in an array
+ * Also, the accumulated result is verified against the expected results
+ * input: Array which has parameter values of different batteries.
+ * returns: Verification of the checked Battery status against the expected.
+ *********************************************************************************/
+int BMS_AttributeStatusAccumulator(float Input_Attribute[NumberOfBatteries][NumberOfParameters],  bool Expected_result[])
+{
+	int VerifyResult=0,VerifiedResult=1;
+	struct BatteryProperties properties;
```

Comment:
This local variable will go out of scope when the function returns, since it is not declared as 'static'